src/http/server/handlers/static_file_handler.cr

```diff
@@ -85,17 +86,18 @@ class HTTP::StaticFileHandler
       if context.request.headers.includes_word?("Accept-Encoding", "gzip")
         gz_file_path = "#{file_path}.gz"
 
-        if File.exists?(gz_file_path) &&
+        if (gz_file_info = File.info?(gz_file_path)) &&
            # Allow small time drift. In some file systems, using `gz --keep` to
            # compress the file will keep the modification time of the original file
            # but truncating some decimals
-           last_modified - modification_time(gz_file_path) < 1.millisecond
+           last_modified - gz_file_info.modification_time < 1.millisecond
           file_path = gz_file_path
+          file_info = gz_file_info
           context.response.headers["Content-Encoding"] = "gzip"
         end
       end
 
-      context.response.content_length = File.size(file_path)
+      context.response.content_length = file_info.not_nil!.size
```

Comment:
What happens if file_info is nil? Can that happen at all?

Reply:
No it can't because the branch is guarded by `is_file` which require `file_info` present. `not_nil!` can still be avoided here, though if we just return early when `file_info` is nil.